src/components/ciPipeline/Webhook/WebhookDetailsModal.tsx

```diff
@@ -0,0 +1,949 @@
+    TOKEN_TAB_LIST,
+} from './webhook.utils'
+import { SchemaType, TabDetailsType, TokenListOptionsType, WebhookDetailsType, WebhookDetailType } from './types'
+import { executeWebhookAPI, getExternalCIConfig, getWebhookAPITokenList } from './webhook.service'
+import Tippy from '@tippyjs/react'
+import { toast } from 'react-toastify'
+import CodeEditor from '../../CodeEditor/CodeEditor'
+import Reload from '../../Reload/Reload'
+
+export function WebhookDetailsModal({ close }: WebhookDetailType) {
+    const { appId, webhookId } = useParams<{
+        appId: string
+        webhookId: string
+    }>()
+    const appStatusDetailRef = useRef<HTMLDivElement>(null)
+    const [loader, setLoader] = useState(false)
+    const [webhookExecutionLoader, setWebhookExecutionLoader] = useState(false)
+    const [generateTokenLoader, setGenerateTokenLoader] = useState(false)
+    const [selectedTokenTab, setSelectedTokenTab] = useState<string>(TOKEN_TAB_LIST[0].key)
+    const [tokenName, setTokenName] = useState<string>('')
+    const [showTokenNameError, setTokenNameError] = useState(false)
+    const [selectedPlaygroundTab, setSelectedPlaygroundTab] = useState<string>(PLAYGROUND_TAB_LIST[0].key)
+    const [selectedRequestBodyTab, setRequestBodyPlaygroundTab] = useState<string>(REQUEST_BODY_TAB_LIST[0].key)
+    const [webhookResponse, setWebhookResponse] = useState<Object>(null)
+    const [selectedToken, setSelectedToken] = useState<TokenListOptionsType>(null)
+    const [generatedAPIToken, setGeneratedAPIToken] = useState<string>(null)
+    const [tokenList, setTokenList] = useState<TokenListOptionsType[]>(undefined)
+    const [showTokenSection, setShowTokenSection] = useState(false)
+    const [isSuperAdmin, setIsSuperAdmin] = useState(false)
+    const [samplePayload, setSamplePayload] = useState<any>(null)
+    const [modifiedSamplePayload, setModifiedSamplePayload] = useState<any>(null)
+    const [sampleJSON, setSampleJSON] = useState(null)
+    const [sampleCURL, setSampleCURL] = useState<any>(null)
+    const [tryoutAPIToken, setTryoutAPIToken] = useState<string>(null)
+    const [showTryoutAPITokenError, setTryoutAPITokenError] = useState(false)
+    const [webhookDetails, setWebhookDetails] = useState<WebhookDetailsType>(null)
+    const [copied, setCopied] = useState(false)
+    const [selectedSchema, setSelectedSchema] = useState<string>('')
+    const [errorInGetData, setErrorInGetData] = useState(false)
+
+    const formatSampleJson = (json): string => {
+        let formattedJSON = ''
+        formattedJSON = JSON.stringify(json, null, 4)
+        return formattedJSON
+    }
+
+    const closeWebhook = (): void => {
+        close()
+    }
+
+    const escKeyPressHandler = (evt): void => {
+        if (evt && evt.key === 'Escape' && typeof close === 'function') {
+            evt.preventDefault()
+            close()
+        }
+    }
+
+    const flattenObject = (ob: Object, tableName: string): Record<string, SchemaType> => {
```

Comment:
Can we use Record type instead of Object?

Reply:
Done